fq.go

```diff
@@ -0,0 +1,60 @@
+package inflight
+
+import (
+	fq "github.com/aaron-prindle/fq-apiserver"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service time
+type FQScheduler struct {
+	fq           *fq.FQScheduler
+	queuedistchs map[int][]interface{}
+}
+
+func NewFQScheduler(queues []fq.FQQueue, clock clock.Clock) *FQScheduler {
+	fq := &FQScheduler{
+		fq:           fq.NewFQScheduler(queues, clock),
+		queuedistchs: make(map[int][]interface{}),
+	}
+	return fq
+}
+
+func (q *FQScheduler) Enqueue(queue fq.FQQueue) <-chan func() {
+	distributionCh := make(chan func(), 1)
+	pkt := &fq.Packet{
+		// TODO(aaron-prindle) HACK, fix
+		QueueIdx: 0,
+		// QueueIdx: queue.Index,
+	}
+	// TODO(aaron-prindle) make it so enqueue fails if the queues are full
+	q.fq.Enqueue(pkt)
+	q.queuedistchs[pkt.QueueIdx] = append(q.queuedistchs[pkt.QueueIdx], distributionCh)
```

Comment:
This costs O(N), where N is the length of the queue.
It would be simpler and more efficient if a Packet contained what we need, whether it be a channel or a condition.